cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageManager.java

```diff
@@ -0,0 +1,96 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import static com.linkedin.openhouse.cluster.storage.StorageType.LOCAL;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * The StorageManager class is responsible for managing the storage types and providing the
+ * appropriate storage implementation based on the configuration.
+ */
+@Component
+public class StorageManager {
+
+  @Autowired StorageProperties storageProperties;
+
+  @Autowired StorageType storageType;
+
+  @Autowired List<Storage> storages;
+
+  /**
+   * Validate the storage properties.
+   *
+   * <p>It validates storage properties as follows:
+   *
+   * <p>1. If any storage type is configured, then default type must be set. Alternatively, if a
+   * default type is not set, then storage types should also be null or empty. **valid**
+   *
+   * <p>2. If default-type is set, then the value of the default type must exist in the configured
+   * storage types. **valid**
+   *
+   * <p>all other configurations are **invalid**
+   */
+  @PostConstruct
+  public void validateProperties() {
+    String clusterYamlError = "Cluster yaml is incorrectly configured: ";
+    if (!StringUtils.hasText(storageProperties.getDefaultType())) {
```

Comment:
Nit: For readability can we do 
if ( condition) .... else
instead of if (!condition) .... else
In this method and also in https://github.com/linkedin/openhouse/pull/90/files#diff-3eab08eaf8a381dc4108e6db8d27ff2f34a126442a904758bb0f5ca9211c54bdR75